crates/primitives/src/transaction/mod.rs

```diff
@@ -1346,6 +1320,179 @@ impl IntoRecoveredTransaction for TransactionSignedEcRecovered {
     }
 }
 
+/// A response to `GetPooledTransactions`. This can include either a blob transaction, or a
+/// non-4844 signed transaction.
+// TODO: redo arbitrary for this encoding - the previous encoding was incorrect
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub enum PooledTransactionsElement {
+    /// A blob transaction, which includes the transaction, blob data, commitments, and proofs.
+    BlobTransaction(BlobTransaction),
+    /// A non-4844 signed transaction.
+    Transaction(TransactionSigned),
```

Comment:
hmm, this now also has eip4844 as a variant, does it support decoding them without the sidecar?

This should be an invalid response, right?

should we flatten the variants?

Reply:
opened this
https://github.com/paradigmxyz/reth/issues/4238